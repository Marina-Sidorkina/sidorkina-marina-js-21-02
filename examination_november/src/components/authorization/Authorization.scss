.authorization {
  position: relative;
  display: flex;
  align-items: center;
  height: 60px;

  &__toggle,
  &__btn {
    display: none;
  }

  @media (max-width: 769px) {
    &__toggle:checked ~ &__btn > &__control {
      transform: rotate(45deg);
    }
    &__toggle:checked ~ &__btn > &__control::before {
      top: 0;
      transform: rotate(0);
    }
    &__toggle:checked ~ &__btn > &__control::after {
      top: 0;
      transform: rotate(90deg);
    }
    &__toggle:checked ~ &__menu {
      display: block;
    }

    &__btn {
      display: flex;
      align-items: center;
      position: absolute;
      top: 20px;
      right: 0;

      width: 26px;
      height: 26px;

      cursor: pointer;
      z-index: 1;
    }

    &__btn > &__control,
    &__btn > &__control::before,
    &__btn > &__control::after {
      display: block;
      position: absolute;

      width: 100%;
      height: 2px;

      background-color: #616161;

      transition-duration: .25s;
    }

    &__btn > &__control::before {
      content: '';
      top: -8px;
    }

    &__btn > &__control::after {
      content: '';
      top: 8px;
    }

    &__menu {
      position: fixed;
      display: none;
      right: 0;
      top: 60px;

      width: 150px;
      height: 100%;

      background-color: #ffffff;
      border-top: 1px solid #f2f2f2;
    }
  }
}
