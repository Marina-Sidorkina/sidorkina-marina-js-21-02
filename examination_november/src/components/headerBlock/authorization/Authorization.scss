.authorization {
  position: relative;
  display: flex;
  align-items: center;
  height: 60px;

  &__toggle,
  &__btn {
    display: none;
  }

  @media (max-width: 769px) {
    &__toggle:checked ~ &__btn > &__control {
      transform: rotate(45deg);
    }
    &__toggle:checked ~ &__btn > &__control::before {
      top: 0;
      transform: rotate(0);
    }
    &__toggle:checked ~ &__btn > &__control::after {
      top: 0;
      transform: rotate(90deg);
    }
    &__toggle:checked ~ &__menu {
      display: block;
    }

    &__toggle:checked ~ &__btn {
      position: fixed;
      right: 15px;
    }

    &__btn {
      display: flex;
      align-items: center;
      position: absolute;
      top: 20px;
      right: 0;

      width: 26px;
      height: 26px;

      cursor: pointer;
      z-index: 10;
    }

    &__btn > &__control,
    &__btn > &__control::before,
    &__btn > &__control::after {
      display: block;
      position: absolute;

      width: 100%;
      height: 2px;

      background-color: #616161;

      transition-duration: .25s;
    }

    &__btn > &__control::before {
      content: '';
      top: -8px;
    }

    &__btn > &__control::after {
      content: '';
      top: 8px;
    }

    &__menu {
      position: fixed;
      display: none;
      right: 0;
      top: 0;
      z-index: 5;

      width: 150px;
      height: 100%;
      padding-top: 60px;

      background-color: #ffffff;
    }
  }
}
