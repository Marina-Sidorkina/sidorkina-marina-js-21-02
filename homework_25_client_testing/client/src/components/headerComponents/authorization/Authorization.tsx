import React from 'react';
import styles from './Authorization.module.scss';
import NotAuthorizedUserBlock from '../notAuthorizedUserBlock/NotAuthorizedUserBlock';
import AuthorizedUserBlock from '../authorizedUserBlock/AuthorizedUserBlock';
import { useTypedSelector } from '../../../redux/hooks/useTypedSelector';

const Authorization = () => {
  const authorizedUserId = useTypedSelector((state) => state.login.data.authorizedUserId);

  return (
    <div className={styles.authorization}>
      <div className={styles.dropdown}>
        <input className={styles.toggle} id="authorization__toggle" type="checkbox" />
        <label className={styles.btn} htmlFor="authorization__toggle">
          <span className={styles.control} />
        </label>
        <div className={styles.menu}>
          { authorizedUserId ? <AuthorizedUserBlock /> : <NotAuthorizedUserBlock />}
        </div>
      </div>
    </div>
  );
};

export default Authorization;
