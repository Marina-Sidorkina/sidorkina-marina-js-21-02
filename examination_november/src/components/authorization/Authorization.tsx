import React from 'react';
import './Authorization.scss';
import AuthorizedUserBlock from '../authorizesUseBlock/AuthorizedUserBlock';

const Authorization = () => (
  <div className="authorization">
    <div className="authorization__dropdown">
      <input className="authorization__toggle" id="authorization__toggle" type="checkbox" />
      <label className="authorization__btn" htmlFor="authorization__toggle">
        <span className="authorization__control" />
      </label>
      <div className="authorization__menu">
        <AuthorizedUserBlock />
      </div>
    </div>
  </div>
);

export default Authorization;
