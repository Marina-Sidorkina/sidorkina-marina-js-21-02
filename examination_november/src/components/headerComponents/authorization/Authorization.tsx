import React from 'react';
import './Authorization.scss';
import { connect } from 'react-redux';
import NotAuthorizedUserBlock from '../notAuthorizedUserBlock/NotAuthorizedUserBlock';
import AuthorizedUserBlock from '../authorizedUserBlock/AuthorizedUserBlock';

interface IAuthorizationProps {
  authorizedUserId: string;
}

const Authorization = (props: IAuthorizationProps) => {
  const { authorizedUserId } = props;
  return (
    <div className="authorization">
      <div className="authorization__dropdown">
        <input className="authorization__toggle" id="authorization__toggle" type="checkbox" />
        <label className="authorization__btn" htmlFor="authorization__toggle">
          <span className="authorization__control" />
        </label>
        <div className="authorization__menu">
          { authorizedUserId ? <AuthorizedUserBlock /> : <NotAuthorizedUserBlock />}
        </div>
      </div>
    </div>
  );
};

export default connect(
  (state: any) => ({
    authorizedUserId: state.login.data.authorizedUserId,
  })
)(Authorization);
