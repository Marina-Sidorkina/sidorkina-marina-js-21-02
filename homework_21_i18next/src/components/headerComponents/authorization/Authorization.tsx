import React from 'react';
import './Authorization.scss';
import NotAuthorizedUserBlock from '../notAuthorizedUserBlock/NotAuthorizedUserBlock';
import AuthorizedUserBlock from '../authorizedUserBlock/AuthorizedUserBlock';
import { useTypedSelector } from '../../../redux/hooks/useTypedSelector';

const Authorization = () => {
  const authorizedUserId = useTypedSelector((state) => state.login.data.authorizedUserId);

  return (
    <div className="authorization">
      <div className="authorization__dropdown">
        <input className="authorization__toggle" id="authorization__toggle" type="checkbox" />
        <label className="authorization__btn" htmlFor="authorization__toggle">
          <span className="authorization__control" />
        </label>
        <div className="authorization__menu">
          { authorizedUserId ? <AuthorizedUserBlock /> : <NotAuthorizedUserBlock />}
        </div>
      </div>
    </div>
  );
};

export default Authorization;
