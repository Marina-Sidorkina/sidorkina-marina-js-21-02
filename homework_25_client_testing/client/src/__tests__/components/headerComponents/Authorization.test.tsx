import React from 'react';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
import { configure, mount } from 'enzyme';
import { Provider } from 'react-redux';
import { HashRouter } from 'react-router-dom';
import mockStore from '../../../../mocks/mockStore';
import Authorization from '../../../components/headerComponents/authorization/Authorization';

configure({ adapter: new Adapter() });

describe('Authorization component test', () => {
  let store: any;
  let wrapper: any;

  beforeEach(() => {
    store = mockStore({
      login: {
        data: {
          authorizedUserId: '123345'
        }
      }
    });
    store.dispatch = jest.fn();
    wrapper = mount(<Provider store={store}><HashRouter><Authorization /></HashRouter></Provider>);
  });

  test('should render Authorization', () => {
    expect(wrapper.length).toEqual(1);
  });

  test('prop should match with store', () => {
    expect(wrapper.prop('login')).toEqual(store.login);
  });

  test('should render LanguageSelect', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
