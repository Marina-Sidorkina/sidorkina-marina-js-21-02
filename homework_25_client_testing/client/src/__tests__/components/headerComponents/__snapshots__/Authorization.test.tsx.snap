// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Authorization component test should render LanguageSelect 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [MockFunction],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <HashRouter>
    <Router
      history={
        Object {
          "action": "POP",
          "block": [Function],
          "createHref": [Function],
          "go": [Function],
          "goBack": [Function],
          "goForward": [Function],
          "length": 1,
          "listen": [Function],
          "location": Object {
            "hash": "",
            "pathname": "/",
            "search": "",
            "state": undefined,
          },
          "push": [Function],
          "replace": [Function],
        }
      }
    >
      <Authorization>
        <div
          className="authorization"
        >
          <div
            className="dropdown"
          >
            <input
              className="toggle"
              id="authorization__toggle"
              type="checkbox"
            />
            <label
              className="btn"
              htmlFor="authorization__toggle"
            >
              <span
                className="control"
              />
            </label>
            <div
              className="menu"
            >
              <AuthorizedUserBlock>
                <div
                  className="userBlock"
                >
                  <div
                    className="item"
                  >
                    <img
                      alt="Фото пользователя"
                      className="userPhoto"
                    />
                    <Link
                      className="link"
                      to="/profile/123345"
                    >
                      <LinkAnchor
                        className="link"
                        href="#/profile/123345"
                        navigate={[Function]}
                      >
                        <a
                          className="link"
                          href="#/profile/123345"
                          onClick={[Function]}
                        >
                          <span
                            className="userName"
                          />
                        </a>
                      </LinkAnchor>
                    </Link>
                  </div>
                  <div
                    className="item"
                  >
                    <span
                      className="logout"
                      onClick={[Function]}
                    >
                      logout
                    </span>
                  </div>
                </div>
              </AuthorizedUserBlock>
            </div>
          </div>
        </div>
      </Authorization>
    </Router>
  </HashRouter>
</Provider>
`;
